botocore/paginate.py

```diff
@@ -191,6 +191,8 @@ def _handle_first_request(self, parsed, primary_result_key,
         # the truncated amount.
         starting_truncation = self._parse_starting_token()[1]
         all_data = primary_result_key.search(parsed)
+        if all_data is None:
+            all_data = []
```

Comment:
Yes I already have a clear idea about what should be adjusted here in order to adapt to the upcoming string support.  I even have a prototype ready.  But it will only work after we merge PR #689 so ideally let's try to merge that one first.